include/Search/TranspositionTable.h

```diff
@@ -99,7 +99,11 @@ struct TranspositionTable
         public:
         static constexpr int NoEval = std::numeric_limits<int16_t>::max();
 
+#ifdef NDEBUG // otherwise used inside the asserts out of the container
```

Comment:
This doesn't look like the greatest idea.
Looks like something that will bite back as a hard to trace bug